From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 3 Sep 2010 16:41:30 -0400
Subject: [net] bnx2x: use pci_ioremap_bar
Message-id: <20100903164130.25415.23258.stgit@brian.englab.brq.redhat.com>
Patchwork-id: 28080
O-Subject: [RHEL5.6 BZ572012 PATCH 05/46] bnx2x: use pci_ioremap_bar
Bugzilla: 572012
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Stanislaw Gruszka <sgruszka@redhat.com>

from upstream 275f165fa970174f8a98205529750e8abb6c0a33

diff --git a/drivers/net/bnx2x_main.c b/drivers/net/bnx2x_main.c
index 4bc1e08..26f97b1 100644
--- a/drivers/net/bnx2x_main.c
+++ b/drivers/net/bnx2x_main.c
@@ -11887,8 +11887,7 @@ static int __devinit bnx2x_init_dev(struct pci_dev *pdev,
 
 	dev->irq = pdev->irq;
 
-	bp->regview = ioremap_nocache(dev->base_addr,
-				      pci_resource_len(pdev, 0));
+	bp->regview = pci_ioremap_bar(pdev, 0);
 	if (!bp->regview) {
 		pr_err("Cannot map register space, aborting\n");
 		rc = -ENOMEM;
